server/odc-service/src/main/java/com/oceanbase/odc/service/flow/task/CreateExternalApprovalTask.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) 2023 OceanBase.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.oceanbase.odc.service.flow.task;
+
+import java.util.Objects;
+import java.util.Optional;
+
+import org.flowable.engine.delegate.DelegateExecution;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import com.oceanbase.odc.common.util.RetryExecutor;
+import com.oceanbase.odc.core.flow.BaseFlowableDelegate;
+import com.oceanbase.odc.service.flow.FlowableAdaptor;
+import com.oceanbase.odc.service.flow.instance.FlowApprovalInstance;
+import com.oceanbase.odc.service.flow.task.model.RuntimeTaskConstants;
+import com.oceanbase.odc.service.flow.util.FlowTaskUtil;
+import com.oceanbase.odc.service.integration.IntegrationService;
+import com.oceanbase.odc.service.integration.client.ApprovalClient;
+import com.oceanbase.odc.service.integration.model.ApprovalProperties;
+import com.oceanbase.odc.service.integration.model.IntegrationConfig;
+import com.oceanbase.odc.service.integration.model.TemplateVariables;
+
+import lombok.extern.slf4j.Slf4j;
+
+/**
+ * @author gaoda.xy
+ * @date 2023/9/1 14:00
+ */
+@Slf4j
+public class CreateExternalApprovalTask extends BaseFlowableDelegate {
+
+    @Autowired
+    private FlowableAdaptor flowableAdaptor;
+    @Autowired
+    private IntegrationService integrationService;
+    @Autowired
+    private ApprovalClient approvalClient;
+    private final RetryExecutor retryExecutor = RetryExecutor.builder().retryIntervalMillis(1000).retryTimes(3).build();
+
+    @Override
+    protected void run(DelegateExecution execution) throws Exception {
+        String activityId = execution.getCurrentActivityId();
+        String processDefinitionId = execution.getProcessDefinitionId();
+        Optional<Optional<Long>> flowInstanceIdOpt = retryExecutor.run(
+                () -> flowableAdaptor.getFlowInstanceIdByProcessDefinitionId(processDefinitionId), Optional::isPresent);
+        if (!flowInstanceIdOpt.isPresent() || !flowInstanceIdOpt.get().isPresent()) {
```

Comment:
what will the flow be when this exception is thrown?

Reply:
The approval node will be terminated, but the flow instance must be terminated by user manually.
The status update is not as expected, I will continue solve this problem.

Reply:
New: If then flowApprovalInstance is not found, then just return and escape the ServiceTask.
If create external approval instance failed, then mark node execute failed and the flow is terminated.